.github/workflows/build_and_test_mac.yml

```diff
@@ -6,41 +6,72 @@ on:
   pull_request: 
   workflow_dispatch:
 
+env:
+  BUILD_TYPE: Release
+
 jobs:
-      
-  Build_And_Run_UnitTest:
-    name: Build and Run Unit Tests
-    runs-on: macos-latest
+  Build_and_Run_GTest:
+    name: Build and Run GoogleTest
+    runs-on: macos-11
+    defaults:
+      run:
+        shell: bash -l {0}
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         submodules: recursive
-
-    - name: Install gcc
-      run: brew install gcc@8
         
-    - name: Update Path
-      run: |
-        echo "brew --prefix llvm)/bin" >> $GITHUB_PATH
-    - name: Setup Python
-      uses: actions/setup-python@v2
+    - uses: conda-incubator/setup-miniconda@v2
       with:
-        python-version: "3.x"
+        activate-environment: anaconda-client-env
+        python-version: 3.8
+        channels: conda-forge
+        auto-activate-base: false
 
-    - name: Install Dependencies
-      run: python3 -m pip install --upgrade wheel
+    - name: Install Conda Dependencies
+      run: |
+        conda install compilers --file ${{github.workspace}}/ci-utils/envs/conda_cpp.txt
 
-    - name: Update setup.py
-      run: python3 -m pip install --upgrade setuptools
+    - name: Configure CMake
+      run: cmake -B ${{github.workspace}} -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DRUN_GTEST=ON -DCMAKE_PREFIX_PATH=$CONDA_PREFIX -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX
+
+    - name: Build
+      run: cmake --build ${{github.workspace}} --config ${{env.BUILD_TYPE}} --parallel 2
+
+    - name: Run GTest Suite
+      working-directory: ${{github.workspace}}
+      # Execute tests.  
+      run: ./tests/cpp/runAllTests  -C ${{env.BUILD_TYPE}}
+      
 
-    - name: Install pytest
-      run: python -m pip install pytest
+  Build_and_Run_PyTest:
+    name: Build and Run PyTest
+    runs-on: macos-11
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive
```

Comment:
We do not want it to be recursive since conda-forge does not yet support git submodules. Rather, we will use the conda dependencies to install pybind11 and modify CMakeLists.txt to detect pybind11 properly.